src/client/Microsoft.Identity.Client/MsalErrorMessage.cs

```diff
@@ -224,6 +224,7 @@ public static string iOSBrokerKeySaveFailed(string keyChainResult)
         public const string TenantIdAndAadAuthorityInstanceAreMutuallyExclusive = "TenantId and AadAuthorityAudience are both set, but they're mutually exclusive. ";
         public const string InstanceAndAzureCloudInstanceAreMutuallyExclusive = "Instance and AzureCloudInstance are both set but they're mutually exclusive. ";
         public const string NoRefreshTokenProvided = "A refresh token must be provided. ";
+        public const string ThrottledTooManyCalls = "You are calling AAD too often and it has throttled you. To avoid this, cache your tokens see https://aka.ms/msal-net-token-caching.";
 
```

Comment:
nit: `AadThrottledError`
and maybe:
`Your app has throttled AAD due to too many requests. See [https://aka.ms/msal-net-throttling](https://aka.ms/msal-net-throttling) for how to avoid this in the future.`

I think you should make a new aka.ms link for throttling because in addition to caching, they should also make sure they are using the right call pattern:
try{
AT Silent}
catch()
{
AT Interactive
}

also, they should not use `WithForceRefresh(true)` 

so all this can be added to one new page and link.

Reply:
We do have this page: [https://aka.ms/msal-net-throttling](https://aka.ms/msal-net-throttling). Maybe the above info an be added there.

This leads to a question - when does AAD return 429 Too Many Calls vs AADSTS50196; or are they the same?

Reply:
Should this be "Your has been throttled by AAD due to too many requests." ? Since AAD is doing the throttling?

Reply:
updated